packages/runtime/src/page.main.ts

```diff
@@ -134,44 +134,6 @@ export const createRoot = (domNode: HTMLElement) => {
     throw new Error('Missing components')
   }
 
-  const urlSignal = window.toddle.locationSignal.map(
-    ({ query, page, route, params, hash }) => {
-      let path: string
-      if (route) {
-        const pathSegments: string[] = []
-        for (const segment of route.path) {
-          if (segment.type === 'static') {
-            pathSegments.push(segment.name)
-          } else {
-            const segmentValue = params[segment.name]
-            if (isDefined(segmentValue)) {
-              pathSegments.push(segmentValue)
-            } else {
-              // If a param is missing, we can't build the rest of the path
-              break
-            }
-          }
-        }
-        path = '/' + pathSegments.join('/')
-      } else {
-        path = compile(page as string, { encode: encodeURIComponent })(params)
-      }
-      const hashString = hash === undefined || hash === '' ? '' : '#' + hash
-      const queryString = Object.entries(query)
-        .filter(([_, q]) => q !== null)
-        .map(([key, value]) => {
-          return `${encodeURIComponent(
-            component?.route?.query[key]?.name ?? key,
```

Comment:
Note that it uses `component` here (which is declared in `pate.main.ts`) - not `route` from the signal. I've changed that during the refactor